vignettes/workflows.Rmd

```diff
@@ -0,0 +1,210 @@
+---
+title: "Workflows with orderly"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{Workflows with orderly}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+`orderly` is designed to support a number of flexible workflows, depending on the size and structure of the group using it.  This vignette describes several workflows, in roughly increasing complexity.
+
+The key issues to consider are
+
+* how will report *sources* be shared?
+* how do the *inputs* (e.g., `orderly` resources) make it into the reports?
+* how will report *outputs* be shared?
+* who's outputs are the "canonical" version?
+
+## Individual researcher
+
+This is the simplest workflow, and the one that `orderly` has the fewest opinions about.  We imagine a single researcher using `orderly` to keep track of a set of analyses.  They would configure orderly with an empty `orderly_config.yml`, and would probably add `archive` to `.gitignore` (`drafts` should always be gitignored).
+
+As the researcher develops reports, they create new directories in `src/` using `orderly::orderly_new()`, run the reports on their own system using `orderly::orderly_run()` and commit them into their `archive` using `orderly::orderly_commit()`.
+
+The researcher needs to take care that the reports are run in a clean R session, as objects and packages in R's global execution environment are available to their reports.
+
+If the researcher wants to share a report with someone else, `orderly` has no opinion on this, and they are free to email a file, or share it in whatever manner they prefer.  The researcher may still quote the report id (e.g., `20200315-100734-10e3de8a`) as a way for them to easily be able to get back to the version of the report that they sent, along with all its source files and resources.
+
+Backup is the responsibility of the researcher; if the computer is backed up, or if the archive lives on a service like Dropbox or OneDrive, this may be sufficient.
+
+### Why not commit the archive to git?
+
+If your reports are very small, then this may work for you.  However, git becomes slow to use when many large files have been added to it, and most hosting platforms, including GitHub will prevent addition of files over a certain size and may complain about overall repositories reaching a certain size.  Cloning the source tree will become increasingly painful, and you may become wary of even *running* a report because of the impact it may have on the above problems.  In our opinion, you should store your archive outside of git and use an alternative method to back it up and distribute it.
+
```

Comment:
Reading this para made me think about the philosophy of reproducible reports...

The source code to generate a single graph might well be far smaller in size than the image file that the code produces. For the source code to create a hundred similar graphs might involve the addition of a couple of lines of loop code, whereas the combined size of the image files the code produces might be a hundred times larger than before. Orderly focuses on storing the code that reproduces a report, rather than storing the report itself, which is a good idea both in terms of keeping storage needs manageable, and also in encouraging the principle of reproducibility in reporting analyses.